packages/amplify-graphql-function-transformer/src/__tests__/amplify-graphql-function-transformer.test.ts

```diff
@@ -40,43 +40,60 @@ test('it generates the expected resources', () => {
         ],
         Version: '2012-10-17',
       },
-    }),
-  );
-  cdkExpect(stack).to(
-    haveResource('AWS::IAM::Policy', {
+    });
+  Template.fromJSON(stack)
+    .hasResourceProperties('AWS::IAM::Policy', {
       PolicyDocument: {
         Statement: [
           {
             Action: 'lambda:InvokeFunction',
             Effect: 'Allow',
-            Resource: {
-              'Fn::If': [
-                'HasEnvironmentParameter',
-                {
-                  'Fn::Sub': ['arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:echofunction-${env}', { env: { Ref: anything() } }],
-                },
-                { 'Fn::Sub': 'arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:echofunction' },
-              ],
-            },
+            Resource: [
+              {
+                'Fn::If': [
+                  'HasEnvironmentParameter',
+                  {
+                    'Fn::Sub': ['arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:echofunction-${env}', { env: { Ref: Match.anyValue() } }],
+                  },
+                  { 'Fn::Sub': 'arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:echofunction' },
+                ],
+              },
+              {
+                'Fn::Join': [
+                  '',
+                  [
+                    {
+                      'Fn::If': [
+                        'HasEnvironmentParameter',
+                        {
+                          'Fn::Sub': ['arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:echofunction-${env}', { env: { Ref: Match.anyValue() } }],
+                        },
+                        { 'Fn::Sub': 'arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:echofunction' },
+                      ],
+                    },
+                    ':*',
+                  ],
+                ],
+              },
+            ],
```

Comment:
This seems to be additive change. I.e. it adds a copy of original rule with trailing `:*` .

I traced the data flow up to this place https://github.com/aws-amplify/amplify-category-api/blob/2c2718284b557dcebb1c8de9f6021722be9ac60b/packages/amplify-graphql-transformer-core/src/transform-host.ts#L344-L350 .
The addition seems to be coming form `LambdaDataSource` which is imported from `@aws-cdk/aws-appsync-alpha`.